src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,123 @@
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @BeforeEach
+    public void setUp1() {
+        user = new User();
+    }
+
+    @Test
+    void register_UserAbove18Age_True() throws RuntimeException {
+        user.setAge(29);
+        user.setPassword("1234567890");
+        user.setLogin("2434349");
+        User expected = user;
+        User actual = registrationService.register(user);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void register_nullAge_notOk() {
+        user.setAge(null);
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_nullPassword_notOk() {
+        user.setPassword(null);
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        user.setLogin(null);
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_UserUnder18Age_notOk() {
+        user.setAge(17);
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_User18Age_notOk() {
+        user.setAge(18);
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_PasswordUnder6Characters_notOk() {
+        user.setPassword("1234");
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_UserNegativeAge_notOk() {
+        user.setAge(-18);
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_UserAlreadyPresent_notOk() throws RuntimeException {
+        user.setAge(23);
+        user.setLogin("23");
+        storageDao.add(user);
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_ReturnUser_True() throws RuntimeException {
+        user.setAge(88);
+        user.setLogin("88");
+        user.setPassword("1234567890");
+        User expected = user;
+        User actual = registrationService.register(user);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void register_ReturnUserId_Ok() throws RuntimeException {
+        user.setAge(83);
+        user.setLogin("83");
+        user.setPassword("1234567890");
+        Long actual = registrationService.register(user).getId();
+        assertNotNull(actual);
+    }
+
+    @Test
+    void register_ReturnLogin_Ok() throws RuntimeException {
+        user.setAge(835);
+        user.setLogin("835");
+        user.setPassword("1234567890");
+        String expected = user.getLogin();
+        String actual = registrationService.register(user).getLogin();
+        assertEquals(expected, actual);
+    }
```

Comment:
These tests are no needed. You have already checked that users are equals and fields are equeals in previous test
```suggestion
```